### PR TITLE
Pixelpaint: Add a Filter Gallery

### PR DESCRIPTION
**Pull request by TobyAsE**
Happy new year everyone!

This PR adds a "Filter Gallery" to PixelPaint.
I'd like Filters to have settings the user can change in the future, which this PR does not achieve, However it is a great step towards supporting that.
All the functionality that lived in the MainWidget regarding Filters moved over to the FilterGallery in the process
(:eye: @mustafaquraish )

image

### CODE REVIEW

File: Userland/Applications/PixelPaint/FilterGallery.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+#include "ImageEditor.h"
+#include <LibGUI/Dialog.h>
+
+namespace PixelPaint {
+
+class FilterGallery final : public GUI::Dialog {
+    C_OBJECT(FilterGallery);
+
+public:
```
Comment (mustafaquraish) [inline Userland/Applications/PixelPaint/FilterGallery.h:17]: Feels a bit weird having an empty `public:` tag here

File: Userland/Applications/PixelPaint/FilterModel.cpp [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+namespace PixelPaint {
+FilterModel::FilterModel(ImageEditor* editor)
+{
+    auto spatial_category = FilterInfo::create_category("Spatial");
+
+    auto edge_detect_category = FilterInfo::create_category("Edge Detect", spatial_category);
+    auto laplace_cardinal_filter = FilterInfo::create_filter(
+        "Laplacian (Cardinal)", [editor]() {
+            (void)editor;
```
Comment (mustafaquraish) [inline Userland/Applications/PixelPaint/FilterModel.cpp:22]: This isn't necessary, is it?

File: Userland/Applications/PixelPaint/FilterModel.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+#include <LibGUI/Model.h>
+
+namespace PixelPaint {
+
+class FilterModel final : public GUI::Model {
+
+public:
+    class FilterInfo : public RefCounted {
+    public:
+        enum Type {
```
Comment (mustafaquraish) [inline Userland/Applications/PixelPaint/FilterModel.h:20]: Perhaps make this an `enum class`? I feel like the code base tends to use those more often.

**Review by mustafaquraish — COMMENTED** [review]
This is much more along the lines of what I had in mind in your earlier PR, seems like a great feature. Some minor comments on the code below.

**Comment by TobyAsE** [comment]
Thanks for the fast review!
The "earlier PR" is over two months ago now, if I remember correctly... But it has been on my mind the whole time

**Pull request merged**